os-boot/Makefile

```diff
@@ -0,0 +1,67 @@
+.PHONY: download_gcc download_linux download_opensbi linux opensbi clean all
+
+all: clean download_gcc download_linux download_opensbi linux opensbi
+
+RISCV_COMPILER_URL ?= https://github.com/riscv-collab/riscv-gnu-toolchain/releases/download/2025.01.20/riscv64-glibc-ubuntu-24.04-gcc-nightly-2025.01.20-nightly.tar.xz
+LINUX_VERSION ?= 6.14.4
+OPENSBI_VERSION ?= 1.6
+
+download_gcc:
+	mkdir -p build
+ifeq (, $(wildcard build/gcc.tar.xz))
+	curl -L $(RISCV_COMPILER_URL) -o build/gcc.tar.xz
+endif
+	curl -L $(RISCV_COMPILER_URL) -o build/gcc.tar.xz
+	tar -C build -xf build/gcc.tar.xz
+
+download_linux:
+	mkdir -p build
+ifeq (, $(wildcard build/linux.tar.xz))
+	curl -L https://cdn.kernel.org/pub/linux/kernel/v6.x/linux-$(LINUX_VERSION).tar.xz -o build/linux.tar.xz
+endif
+	tar -C build -xf build/linux.tar.xz
+
+download_opensbi:
+	mkdir -p build
+ifeq (, $(wildcard build/opensbi.tar.xz))
+	curl -L https://github.com/riscv-software-src/opensbi/archive/refs/tags/v$(OPENSBI_VERSION).tar.gz -o build/opensbi.tar.xz
+endif
+	tar -C build -xf build/opensbi.tar.xz
+
+CROSS_COMPILE := $(shell pwd)/build/riscv/bin/riscv64-unknown-linux-gnu-
+
+elf := build/opensbi-$(OPENSBI_VERSION)/build/platform/generic/firmware/fw_payload.elf
+
+# Number of instructions to run.
+LIMIT_INSTRUCTIONS ?= 250000
+
+linux:
```

Comment:
Would be nice if we made the dependency tree work. E.g. this should depend `build/linux-$(LINUX_VERSION)/Makefile` and the `download_linux` rule should be

```
build/linux-$(LINUX_VERSION)/Makefile:
    mkdir -p build
...
```

I can do it if you like when the other comments are resolved.

Reply:
I will do it after no one objects in other comments after I finish my roommate's birthday lunch.